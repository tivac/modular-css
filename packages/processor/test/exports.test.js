"use strict";

const dedent = require("dedent");
const namer  = require("@modular-css/test-utils/namer.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("exports", () => {
        let processor;
        
        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });
        
        it("should export an object of arrays containing strings", async () => {
            const { exports } = await processor.string(
                "./simple.css",
                dedent(`
                    .red { color: red; }
                    .black { background: #000; }
                    .one, .two { composes: red, black; }
                `)
            );

            expect(exports).toMatchSnapshot();
        });

        it("should export identifiers and their classes", async () => {
            await processor.file(
                "./packages/processor/test/specimens/start.css"
            );

            const { compositions } = await processor.output();
            
            expect(compositions).toMatchSnapshot();
        });
    });
});
