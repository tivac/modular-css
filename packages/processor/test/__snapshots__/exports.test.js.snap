// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js exports should export an object of arrays containing strings 1`] = `
Object {
  "black": Array [
    "black",
  ],
  "one": Array [
    "red",
    "black",
    "one",
  ],
  "red": Array [
    "red",
  ],
  "two": Array [
    "red",
    "black",
    "two",
  ],
}
`;

exports[`/processor.js exports should export identifiers and their classes 1`] = `
Object {
  "packages/processor/test/specimens/folder/folder.css": Object {
    "folder": Array [
      "folder",
    ],
  },
  "packages/processor/test/specimens/local.css": Object {
    "booga": Array [
      "booga",
    ],
    "looga": Array [
      "booga",
      "looga",
    ],
  },
  "packages/processor/test/specimens/start.css": Object {
    "booga": Array [
      "booga",
    ],
    "tooga": Array [
      "tooga",
    ],
    "wooga": Array [
      "booga",
      "wooga",
    ],
  },
}
`;
